python/tskit/tables.py

```diff
@@ -2514,3 +2514,41 @@ def subset(self, nodes, record_provenance=True):
             self.provenances.add_row(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
+
+    def union(self, other, nodes, record_provenance=True):
+        """
+        Modifies the tables in place to contain only the entries referring to
+        the provided list of nodes, with nodes reordered according to the order
+        they appear in the list. See :meth:`TreeSequence.subset` for a more
+        detailed description.
+
+        :param list nodes: The list of nodes for which to retain information. This
+            may be a numpy array (or array-like) object (dtype=np.int32).
+        :param bool record_provenance: Whether to record a provenance entry
+            in the provenance table for this operation.
+        """
+        nodes = util.safe_np_int_cast(nodes, np.int32)
+        self.ll_tables.union(other.ll_tables, nodes)
+        if record_provenance:
+            other_records = [prov.record for prov in other.provenances]
+            other_timestamps = [prov.timestamp for prov in other.provenances]
+            parameters = {
+                "command": "union",
+                "other": {"timestamp": other_timestamps, "records": other_records},
+                "nodes": nodes.tolist(),
+            }
+            self.provenances.add_row(
+                record=json.dumps(provenance.get_provenance_dict(parameters))
+            )
+
+    def add_time(self, dt):
```

Comment:
there's no tests for this - did you mean to leave this in?

Reply:
idk what to do with this. Do we want to make this an official method or not?

Reply:
If we need it for our use case (and, I think we do?) then I think we should. If we don't need it, then no.

Reply:
We don't because our species tree is ultrametric (unless I f* something up in the SLiM sims heh). I could move this to a separate issue and in the future we or someone else can make the tests and description.

Reply:
meh, if we don't need it then just forget about it, I'd say. we don't want to clutter up our issues/PR pages with things we don't actually intend to do.